### PR TITLE
feat: extract component classes

### PR DESCRIPTION
**Pull request by plagoa**

### CODE REVIEW

**Review by francisco-guilherme — APPROVED** [review]

File: packages/core/src/components/AppSwitcher/AppSwitcher.tsx [hunk 1]
```diff
@@ -123,6 +110,7 @@ export const HvAppSwitcher = ({
       id={id}
       className={clsx(
         className,
+        appSwitcherClasses.root,
         classes?.root,
         isOpen && classes?.open,
         isOpen === false && classes?.closed
```
Comment (MEsteves22) [inline packages/core/src/components/AppSwitcher/AppSwitcher.tsx:116]: Missing open and closed classes.

File: packages/core/src/components/BaseInput/BaseInput.tsx [hunk 2]
```diff
@@ -137,7 +111,7 @@ export const HvBaseInput = ({
   return (
     <StyledRoot
       className={clsx(
-        "root",
+        baseInputClasses.root,
         classes?.root,
         className,
         formElementProps.disabled && classes?.disabled,
```
Comment (MEsteves22) [inline packages/core/src/components/BaseInput/BaseInput.tsx:117]: Classes disabled, invalid, resizable, and readOnly missing.

File: packages/core/src/components/EmptyState/EmptyState.tsx [hunk 3]
```diff
@@ -54,19 +45,40 @@ export const HvEmptyState = (props: HvEmptyStateProps) => {
...
   return (
-    <StyledRoot className={className} {...others}>
+    <StyledRoot
+      className={clsx(className, emptyStateClasses.root, classes?.root)}
+      {...others}
+    >
       <StyledContainer
         breakpoints={muiTheme.breakpoints}
         messageOnly={!!(message && !(title || action))}
       >
```
Comment (MEsteves22) [inline packages/core/src/components/EmptyState/EmptyState.tsx:55]: Missing icon and iconContainer classes.

File: packages/core/src/components/Input/Input.tsx [hunk 4]
```diff
@@ -154,46 +155,7 @@ export type HvInputProps = HvBaseProps<
...
-    /** Styles applied to the icon used to clean the input. */
-    iconClear?: string;
-    /** Styles applied to the input extension shown when the suggestions list is visible. */
-    inputExtension?: string;
-    /** Styles applied to the container of the suggestions list. */
-    suggestionsContainer?: string;
-    /** Styles applied to the suggestions list. */
-    suggestionList?: string;
-  };
+  classes?: HvInputClasses;
```
Comment (MEsteves22) [inline packages/core/src/components/Input/Input.tsx:158]: Missing classes input, inputRoot, inputRootFocused, inputRootDisabled, inputRootMultiline, and inputBorderContainer in `StyledBaseInput` component.

File: packages/core/src/components/SelectionList/SelectionList.tsx [hunk 5]
```diff
@@ -291,7 +283,7 @@ export const HvSelectionList = ({
       disabled={disabled}
       required={required}
       readOnly={readOnly}
-      className={clsx(className, classes?.root)}
+      className={clsx(className, selectionListClasses.root, classes?.root)}
     >
       {label && (
         <StyledLabel
```
Comment (MEsteves22) [inline packages/core/src/components/SelectionList/SelectionList.tsx:289]: Missing label class.

File: packages/core/src/components/Tag/Tag.tsx [hunk 6]
```diff
@@ -37,21 +38,7 @@ export type HvTagProps = Omit<MuiChipProps, "color"> &
...
-      chipRoot?: string;
-      categorical?: string;
-      disabled?: string;
-      clickable?: string;
-      categoricalFocus?: string;
-      categoricalDisabled?: string;
-      deleteIcon?: string;
-      disabledDeleteIcon?: string;
-    };
+    classes?: HvTagClasses;
```
Comment (MEsteves22) [inline packages/core/src/components/Tag/Tag.tsx:41]: Classes tagButton, focusVisible, and primaryButton missing in `StyledButton`.

File: packages/core/src/components/TagsInput/TagsInput.tsx [hunk 7]
```diff
@@ -535,7 +489,8 @@ export const HvTagsInput = ({
       onBlur={onBlurHandler}
       onFocus={onFocusHandler}
       className={clsx(
-        "root",
+        // "root",
+        tagsInputClasses.root,
         classes?.root,
         className,
         disabled && classes?.disabled
```
Comment (MEsteves22) [inline packages/core/src/components/TagsInput/TagsInput.tsx:496]: Class disabled missing.

File: packages/core/src/components/TagsInput/TagsInput.tsx [hunk 8]
```diff
@@ -545,7 +500,7 @@ export const HvTagsInput = ({
         <StyledLabelContainer className={classes?.labelContainer}>
```
Comment (MEsteves22) [inline packages/core/src/components/TagsInput/TagsInput.tsx:500]: Class labelContainer missing.

File: packages/core/src/components/TagsInput/TagsInput.tsx [hunk 9]
```diff
@@ -576,7 +537,8 @@ export const HvTagsInput = ({
 
       <StyledTagsList
         className={clsx(
-          "tagsList",
+          // "tagsList",
+          tagsInputClasses.tagsList,
           classes?.tagsList,
```
Comment (MEsteves22) [inline packages/core/src/components/TagsInput/TagsInput.tsx:542]: Classes error, resizable, invalid, and singleLine missing.

File: packages/core/src/components/TagsInput/TagsInput.tsx [hunk 10]
```diff
@@ -607,7 +569,10 @@ export const HvTagsInput = ({
...
                 key={`${tag.label}-${i}`}
                 tabIndex={-1}
-                className={clsx(!multiline && classes?.singleLine)}
+                className={clsx(
+                  !multiline &&
+                    clsx(tagsInputClasses.singleLine, classes?.singleLine)
+                )}
                 classes={{
                   gutters: classes?.listItemGutters,
                   root: classes?.listItemRoot,
```
Comment (MEsteves22) [inline packages/core/src/components/TagsInput/TagsInput.tsx:578]: Classes listItemGutters and listItemRoot missing.

File: packages/core/src/components/TagsInput/TagsInput.tsx [hunk 11]
```diff
@@ -617,7 +582,10 @@ export const HvTagsInput = ({
               >
                 <StyledTag
                   label={label}
-                  className={clsx(i === tagCursorPos && classes?.tagSelected)}
+                  className={clsx(
+                    i === tagCursorPos &&
+                      clsx(tagsInputClasses.tagSelected, classes?.tagSelected)
+                  )}
                   classes={{
                     chipRoot: classes?.chipRoot,
```
Comment (MEsteves22) [inline packages/core/src/components/TagsInput/TagsInput.tsx:590]: Class chipRoot missing.

File: packages/core/src/components/TagsInput/TagsInput.tsx [hunk 12]
```diff
@@ -637,15 +605,17 @@ export const HvTagsInput = ({
...
             className={clsx(
-              !multiline && classes?.singleLine,
-              value.length === 0 ? "empty" : ""
+              !multiline &&
+                clsx(tagsInputClasses.singleLine, classes?.singleLine),
+              value.length === 0 ? tagsInputClasses.tagInputRootEmpty : ""
             )}
             classes={{
               root: classes?.tagInputContainerRoot,
               gutters: classes?.listItemGutters,
```
Comment (MEsteves22) [inline packages/core/src/components/TagsInput/TagsInput.tsx:614]: Classes tagInputContainerRoot and listItemGutters missing.

File: packages/core/src/components/TagsInput/TagsInput.tsx [hunk 13]
```diff
@@ -676,7 +649,6 @@ export const HvTagsInput = ({
               }}
               inputRef={inputRef}
               $singleLine={!multiline}
-              $isTagSelected={isTagSelected}
               {...others}
             />
           </StyledInputListItem>
```
Comment (MEsteves22) [inline packages/core/src/components/TagsInput/TagsInput.tsx:654]: Classes inputExtension, suggestionsContainer, and suggestionList missing for the suggestion component.

**Pull request merged**